engine/src/main/java/com/redhat/cloud/notifications/processors/ConnectorSender.java

```diff
@@ -64,11 +77,31 @@ public void send(Event event, Endpoint endpoint, JsonObject payload) {
 
         notificationHistoryRepository.createNotificationHistory(history);
 
+        // Measure the payload size.
         final int payloadSize = payload.toString().getBytes().length;
         recordMetrics(event, connector, payloadSize);
 
+        final Map<String, String> customHeaders = new HashMap<>();
+
+        // When the payload to be sent is greater than the configured limit,
+        // store the payload in the database so that we can fetch it from the
+        // connectors themselves.
+        if (this.engineConfig.getKafkaToCamelMaximumRequestSize() <= payloadSize) {
```

Comment:
Hmmm but we're transforming the String to an array of bytes and getting that length. That should give us the size in bytes of the payload, roughly, no? Maybe we can apply the `UTF-8` encoding to make sure, in case we start using emojis in payloads...